src/HistoManager.cpp

```diff
@@ -122,6 +158,27 @@ void HistoManager::AddGenInfo(VtxInformation* info)
       
 }
 
+void HistoManager::AddGenInfoParticles(G4PrimaryParticle* particle)
+{
+  PrimaryParticleInformation* infoParticle = static_cast<PrimaryParticleInformation*> (particle->GetUserInformation());
+
+  if (infoParticle == nullptr) return;
+
+  G4int multiplicity = infoParticle->GetGammaMultiplicity();
+
+  if ( multiplicity == 0 ) return;
+
+  if ( multiplicity == 1 )
+  {
+    fGeantInfo->SetMomentumGamma0( particle->GetPx(), particle->GetPy(), particle->GetPz());
+  } else {
+    G4int index = infoParticle->GetIndex();
+    if (index==1) fGeantInfo->SetMomentumGamma1( particle->GetPx(), particle->GetPy(), particle->GetPz());
+    if (index==2) fGeantInfo->SetMomentumGamma2( particle->GetPx(), particle->GetPy(), particle->GetPz());
+    if (index==3) fGeantInfo->SetMomentumGamma3( particle->GetPx(), particle->GetPy(), particle->GetPz());
+  }
```

Comment:
maybe represent index with enums?

Reply:
Here I need more explanation - now I can not see how this will simplify the code